R/LSTdecisionTree.R

```diff
@@ -0,0 +1,278 @@
+                                  "Bootstrapped t-test or \n Wilcoxon matched-pairs \n test",
+                                  "Bootstrapped t-test or \n Mann-Whitney Test", "Bootstrapped ANOVA or\nFriedman's ANOVA",
+                                  "Robust ANOVA or \n Kruskal-Wallis test", "Bootstrapped multiple\nregression",
+                                  "Robust factorial repeated\nmeasures ANOVA", "Robust independent \n factorial ANOVA/\n multiple regression",
+                                  "Robust factorial mixed\nANOVA", "Robust ANCOVA/\nbootstrapped regression"))
+  graph <- igraph::graph_from_data_frame(decisionDf, directed = TRUE)
+  coords <- igraph::layout_as_tree(graph)
+  names <- igraph::vertex_attr(graph, "name")
+  
+  # data frame for nodes and box size
+  boxHeight <- 0.46
+  boxWidth <-  .3
+  nodesDf <- data.frame(label = names, x = coords[,2], y = coords[,1], 
+                        xmin = coords[,2] - boxWidth, ymin = coords[,1] - boxHeight,
+                        xmax = coords[,2] + boxWidth, ymax = coords[,1] + boxHeight)
+  
+  # remove NULL nodes from edgesDf and decision Df
+  nodesDf <- subset(nodesDf, !grepl("NULL", nodesDf$label))
+  decisionDf <- subset(decisionDf, !(grepl("NULL", decisionDf$from) | grepl("NULL", decisionDf$to)))
+  
+  # add edges that skip rows
+  extraEdges <- data.frame(from = c("Continuous3", "Continuous4", "Categorical3", "Categorical3", "Categorical3",
+                                    "Both", "Continuous5", "Categorical4", "Continuous6", "Categorical5", "Both2",
+                                    "Categorical6", "Categorical7", "Both3"),
+                           to = c("Pearson correlation \n or regression", "Multiple regression", "Same3", "Different3",
+                                  "Both4", "Multiple regression\n/ANCOVA", "Logistic regression or \n biserial/point biserial \n correlation",
+                                  "Different4", "Logistic regression", "Different5", "Different6", "MANOVA", "Factorial MANOVA", "MANCOVA"))
+  decisionDf <- rbind(decisionDf, extraEdges)
+  
+  # data frame for edges
+  edgesDf <- cbind(decisionDf, "id" = seq_len(nrow(decisionDf))) 
+  edgesDf <- tidyr::pivot_longer(edgesDf, cols = c("from", "to"),
+                                 names_to = "order",
+                                 values_to = "label")
+  xCoord <- c()
+  yCoord <- c()
+  for (i in seq_len(nrow(edgesDf))) {
+    yCoord <- c(yCoord, nodesDf$y[nodesDf$label == edgesDf$label[i]])
+    xDirection <- edgesDf$order[i]
+    newXCoord <- ifelse(xDirection == "from", nodesDf$xmin[nodesDf$label == edgesDf$label[i]], nodesDf$xmax[nodesDf$label == edgesDf$label[i]])
+    xCoord <- c(xCoord, newXCoord)
+  }
+  edgesDf <- cbind(edgesDf, x = xCoord, y = yCoord)
+  
+  # set active nodes
+  ## create list with all paths to parametric tests
+  allTestNames <- tail(nodesDf, 28)$label
```

Comment:
If 28 == nrow(decisionDf), please just write that instead. Otherwise, where does the 28 come from?

Reply:
There are 28 tests in the decision tree of Field, these are the final nodes (i.e., tail(28)) of the graph. Maybe there is a better way to select them, but the decision tree is also fixed. Or should I rewrite it in a way that makes it easier to make deviations from the Field decision tree?